packages/react-async/src/hooks.ts

```diff
@@ -25,7 +25,7 @@ export type KeepFreshable<KeepFreshOptions extends object> = Pick<
 export function usePreload<PreloadOptions extends object>(
   ...args: IfAllOptionalKeys<
     PreloadOptions,
-    [Preloadable<PreloadOptions>, NoInfer<PreloadOptions>?],
+    [Preloadable<PreloadOptions>, PreloadOptions?],
```

Comment:
If you hack in these type changes locally into web's typing for `react-async` does the codebase still type-check successfully?

Reply:
I tried it out with the tophat command and `dev type-check` passes